src/app/content-blocks/block-news/block-news.component.ts

```diff
@@ -61,6 +60,11 @@ export class BlockNewsComponent implements OnInit {
               this.articles = this.articles.filter(a => a.tags && this.content.filterByTag.split(',')
                 .every(tag => a.tags.indexOf(tag) > -1));
             }
+            if (this.content && this.content.frontpageFeaturedNewsOnly) {
+              this.articles = this.articles.filter((article) => {
+                return article.onFrontpage === true;
+              });
+            }
```

Comment:
This could have been done as graphql query param, your version works too, but cleaner version would be:

`if (this.content && this.content.frontpageFeaturedNewsOnly) {
      params['fields.onFrontpage '] = true;
 }`

Added to line 30, same way we already filter articles by categories. This way we don't need to filter articles in frontend